### PR TITLE
Don't assume that single kind topics won't have duplicate slugs.

### PR DESCRIPTION
**Pull request by rtibbles**
This will require a rerun of contentload, but not for assessment items.

### CODE REVIEW

**Comment by rtibbles** [comment]
Fixes #4396

**Comment by rtibbles** [comment]
Also removes items not in the topic tree from content caches.

Reran contentload too to update content caches.

```
INFO:kalite:Downloaded topic_tree data for
            1806 exercises
            7393 content files
```

File: kalite/contentload/management/commands/channels/khan.py [hunk 1]
```diff
@@ -139,6 +139,12 @@ def build_full_cache(items, id_key="id"):
         except AttributeError:
             logging.error("Unable to serialize %r" % item)
 
+        item = whitewash_node_data(item)
+
+        if ids:
+            if item["id"] not in ids:
```
Comment (jamalex) [inline kalite/contentload/management/commands/channels/khan.py]: Should this use `id_key` (or what's that for?) -- and is `ids` a blacklist? May be a better name for it?

File: kalite/contentload/management/commands/contentload.py [hunk 2]
```diff
@@ -37,6 +37,20 @@ def scrub_topic_tree(node_cache, channel_data):
                     del node[att]
 
 
+def topic_tree_id_catalog(topic_tree):
+    # Takes the topic tree and returns a dictionary with the ids of all nodes in the tree
+    # This is for use in removing items from caches that are not in the topic tree
```
Comment (jamalex) [inline kalite/contentload/management/commands/contentload.py]: Docstring?

File: kalite/contentload/management/commands/contentload.py [hunk 3]
```diff
@@ -37,6 +37,20 @@ def scrub_topic_tree(node_cache, channel_data):
...
+    ids = {}
```
Comment (jamalex) [inline kalite/contentload/management/commands/contentload.py]: Why not just a set...?

File: kalite/contentload/management/commands/channels/khan.py (hunk 1 above)
Comment (rtibbles) [inline kalite/contentload/management/commands/channels/khan.py]: I think id_key is actually redundant at this point, it was being used previously to double check things prior to whitewash_node_data being applied, but after whitewash_node_data, every node has 'id' as its id_key.

**Pull request merged**